deploy/Dockerfile

```diff
@@ -17,31 +17,56 @@
 FROM golang:1.14
 ARG GOARCH=amd64
 WORKDIR /go/src/github.com/GoogleContainerTools/kaniko
+
+RUN echo $GOARCH > /goarch
+
+#This arg is passed by docker buildx & contains the platform info in the form linux/amd64, linux/ppc64le etc.
+ARG TARGETPLATFORM
```

Comment:
you could use `TARGETARCH` instead

Reply:
As per https://docs.docker.com/buildx/working-with-buildx/ , TARGETPLATFORM is automatically passed and hence used same.